lib/entry.js

```diff
@@ -257,6 +259,23 @@ function forEachSetter(entry, names, callback) {
 
       } else {
         call(setters[key], name, value, callback);
+
+        var getter = entry.getters[name];
+        if (typeof getter === "function" &&
+            // Sometimes a getter function will throw because it's called
+            // before the variable it's supposed to return has been
+            // initialized, so we need to know that the getter function
+            // has run to completion at least once.
```

Comment:
Added a test that I think demonstrates this `getter.runCount > 0` check matters, even though the test itself is a bit artificial.